src/faebryk/libs/sets.py

```diff
@@ -398,14 +398,14 @@ def __init__(
         self.units = units or min_unit or max_unit
         self.range_units = base_units(self.units)
 
-        if isinstance(min, Quantity):
+        if HasUnit.check(min):
             num_min = min.to_base_units().magnitude
```

Comment:
This can not work. HasUnit.check checks if something has an units attr, but in the following we act like the variable is a Quantity (.to_base_units)